pkg/sparrow/run.go

```diff
@@ -41,16 +42,26 @@ type Sparrow struct {
 	db db.DB
 	// the existing checks
 	checks map[string]checks.Check
+	server *http.Server
 
 	metrics Metrics
 
 	resultFanIn map[string]chan checks.Result
-	cResult     chan checks.ResultDTO
+	cfg         *config.Config
 
-	cfg        *config.Config
-	loader     config.Loader
+	// cCfgChecks is the channel where the loader sends the runtime configuration of the checks
 	cCfgChecks chan map[string]any
-	targets    targets.TargetManager
+	// cResult is the channel where the checks send their results to
+	cResult chan checks.ResultDTO
+	// cErr is used to handle non-recoverable errors of the sparrow components
+	cErr chan error
+	// cDone is used to signal that the sparrow was shut down because of an error
+	cDone chan struct{}
+	// shutOnce is used to ensure that the shutdown function is only called once
+	shutOnce sync.Once
```

Comment:
Imo the sparrow struct is already too overloaded. I think it'd be better to split the fields into multiple structs to reduce complexity and improve readability and then create a composition in the sparrow struct by embedding the new smaller structs. 

e.g.:
```go
 type Sparrow struct {
	Components
	Errors
	Checks

	db db.DB

	metrics Metrics
}

type errHandlers struct {
	// cErr is used to handle non-recoverable errors of the sparrow components
	cErr chan error
	// cDone is used to signal that the sparrow was shut down because of an error
	cDone chan struct{}
	// shutOnce is used to ensure that the shutdown function is only called once
	shutOnce sync.Once
}

type components struct {
	loader config.Loader

	tarMan targets.TargetManager

	server      *http.Server
	routingTree *api.RoutingTree
	router      chi.Router
}

type checkComponents struct {
	// checks includes the existing checks
	checks      map[string]checks.Check
	resultFanIn map[string]chan checks.Result
	cfg         *config.Config

	// cCfgChecks is the channel where the loader sends the runtime configuration of the checks
	cCfgChecks chan map[string]any
	// cResult is the channel where the checks send their results to
	cResult chan checks.ResultDTO
}
```

But as this shouldn't be part of this PR, I'd recommend you to just put the new err handling fields into a new struct.

Reply:
I was thinking of something like this as well. Let's create an issue an handle this at a later point. The api should also be it's own component, for example.

I'd prefer to leave it as is, as we're already changing quite a few things here. Let's try to restrain ourselves and take it step by step 😉

The sparrow struct is already too big, so just a couple of concurrency mumbo jumbo is OK, IMO.